testplan/monitor/resource.py

```diff
@@ -106,15 +134,28 @@ def collect_cpu_usage(self) -> float:
     def collect_memory_usage(self) -> int:
         return self.memory_size - psutil.virtual_memory().available
 
+    @staticmethod
+    def _ensure_positive(num):
```

Comment:
This is a bit misleading. We ensure non-negative not positive. Also I think it is cleaner to write return max(x, 0). Third thing: do you need to ensure integer as well?